libraries/persistent_store/src/format.rs

```diff
@@ -1077,4 +1077,12 @@ mod tests {
             0xff800000
         );
     }
+
+    #[test]
+    fn position_offsets_fit_in_a_halfword() {
+        // The store stores the entry positions as their offset from the head. Those offsets are
+        // represented as u16. The bound below is a large over-approximation of the maximal offset
+        // but it already fits.
+        assert_eq!((MAX_PAGE_INDEX + 1) * MAX_VIRT_PAGE_SIZE, 0xff80);
```

Comment:
According to the name of the test, shouldn't you check that the computed value is less or equal than `u16::MAX` then?

Reply:
Yes, but this is a consequence of the test. The 2 alternatives are:
1. `assert_eq!(BOUND, 0xff80)` which implies `BOUND <= u16::MAX`
2. `assert!(BOUND <= u16::MAX)`

I think the first is stable enough that it should not increase maintenance while documenting the actual maximum value (which fits in `u16` which is the goal of the test).

Actually there's a 3rd option if you prefer:
```rust
const MAX_POS: Nat = (MAX_PAGE_INDEX + 1) * MAX_VIRT_PAGE_SIZE;
assert_eq!(MAX_POS, 0xff80);  // documentation
assert!(MAX_POS <= u16::MAX as Nat);  // redundant but what actually matters
```

Reply:
I think the third option makes more sense provided you swap the 2 assertions:
```rust
const MAX_POS: Nat = (MAX_PAGE_INDEX + 1) * MAX_VIRT_PAGE_SIZE;
assert!(MAX_POS <= u16::MAX as Nat);  // what actually matters
assert_eq!(MAX_POS, 0xff80);  // redundant but documentation
```

Reply:
Sounds good. Done.